src/lib/cmdgen.c

```diff
@@ -718,21 +720,35 @@ char *lyd2cmdline_args(const struct lyd_node *startcmd_node, oper_t op_val,
                         __func__, next->schema->name);
                 return NULL;
             }
+            int is_tail_arg = 0;
+            if (get_extension(ADD_LEAF_AT_END, next, NULL) == EXIT_SUCCESS)
+                is_tail_arg = 1;
             if (arg_name != NULL) {
-                strlcat(cmd_line, " ", sizeof(cmd_line));
-                strlcat(cmd_line, arg_name, sizeof(cmd_line));
+                if (is_tail_arg) {
+                    strlcat(tail_arg, " ", sizeof(tail_arg));
+                    strlcat(tail_arg, arg_name, sizeof(tail_arg));
+                } else {
+                    strlcat(cmd_line, " ", sizeof(cmd_line));
+                    strlcat(cmd_line, arg_name, sizeof(cmd_line));
+                }
                 free(arg_name);
             }
             if (arg_value != NULL) {
-                strlcat(cmd_line, " ", sizeof(cmd_line));
-                strlcat(cmd_line, arg_value, sizeof(cmd_line));
+                if (is_tail_arg) {
+                    strlcat(tail_arg, " ", sizeof(tail_arg));
+                    strlcat(tail_arg, arg_value, sizeof(tail_arg));
+                } else {
+                    strlcat(cmd_line, " ", sizeof(cmd_line));
+                    strlcat(cmd_line, arg_value, sizeof(cmd_line));
+                }
                 free(arg_value);
             }
             break;
         }
         LYD_TREE_DFS_END(startcmd_node, next);
     }
 done:
+    strlcat(cmd_line, tail_arg, sizeof(cmd_line)); // add the tail arg to the line.
     return strdup(cmd_line);
```

Comment:
maybe not for this PR, but I noticed we never free cmd_line

Reply:
Cmd_line in stack memory.